pkg/configuration/user/casc/casc.go

```diff
@@ -45,15 +45,30 @@ func (c *configurationAsCode) Ensure(jenkins *v1alpha2.Jenkins) (requeue bool, e
 }
 
 const applyConfigurationAsCodeGroovyScriptFmt = `
-String[] configContent = ['''%s'''] 
+import hudson.util.VersionNumber
+
+String[] configContent = ['''%s''']
 def configSb = new StringBuffer()
 for (int i=0; i<configContent.size(); i++) {
     configSb << configContent[i]
 }
 
 def stream = new ByteArrayInputStream(configSb.toString().getBytes('UTF-8'))
 
-def source = new io.jenkins.plugins.casc.yaml.YamlSource(stream, io.jenkins.plugins.casc.yaml.YamlSource.READ_FROM_INPUTSTREAM)
+def plugin = jenkins.model.Jenkins.instance.getPluginManager().whichPlugin(io.jenkins.plugins.casc.ConfigurationAsCode)
+def version = plugin.getVersionNumber()
+
+def source
+
+switch (version) {
+	case { it.isNewerThanOrEqualTo(new VersionNumber("1.41")) }:
+		source = new io.jenkins.plugins.casc.yaml.YamlSource(stream)
+		break
+	default:
+		source = new io.jenkins.plugins.casc.yaml.YamlSource(stream, io.jenkins.plugins.casc.yaml.YamlSource.READ_FROM_INPUTSTREAM)
+		break
+	}
+
 io.jenkins.plugins.casc.ConfigurationAsCode.get().configureWith(source)
```

Comment:
Here is a clearer version that is backwards compatible.

```suggestion
def source = io.jenkins.plugins.casc.yaml.YamlSource.of(stream)
io.jenkins.plugins.casc.ConfigurationAsCode.get().configureWith(source)
```

However this method is marked restricted for a reason. https://github.com/jenkinsci/configuration-as-code-plugin/blob/1c32633d879379b83bb4c0475bf9876c8f88c051/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L606-L611

If you need an official way to call with stdin we currently offer the [`ApplyConfigurationCommand`](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/casc/cli/ApplyConfigurationCommand.java) That will takes `stdin`.

If you need a better entrypoint you could have asked for it when creating this.

Reply:
As the one who made the so called breaking change.
I do not think so.

You were calling the wrong API in the first place.
This was the very reason why we offer the `YamlSource.of()` method, so that the internal method could change.
This pattern was removed to avoid leaking readers that were holding file handles.

See this PR: https://github.com/jenkinsci/configuration-as-code-plugin/pull/1398

And see specifially this comment: https://github.com/jenkinsci/configuration-as-code-plugin/pull/1398#discussion_r434237047

Reply:
@jetersen Thanks for the detailed explanation! And sorry for not reaching out before creating the PR, my apologies for rushing it, I should have done my due diligence. 

The two lines are certainly nicer than what I ended up with😰 I do understand your concern with calling `configureWith`. I am not familiar enough with the operator and jcasc code base to suggest an alternative to the original implementation, so this issue is better handled by someone else than me. 

I am very green when it comes to contributing to any project, so I am unsure what a good follow up would be, my suggest is to close this PR and open an issue regarding how the configurations is applied, does that sound like the right next step to you?

Reply:
I think the next should be applying my suggestions. Than the maintainers can merge this PR.
Credit where credit is due. I am very thankful that you found this issue.

Than we can consider our next time in another issue or PR.

Reply:
👍 I applied you suggestion, and thanks again for your feedback. I am just excited to start using anchors in my configuration😀